CMakeLists.txt

```diff
@@ -22,6 +22,14 @@ ENDIF(COMMAND CMAKE_POLICY)
 # Import of windows.h breaks min()/max()
 ADD_DEFINITIONS(-DNOMINMAX)
 
+# Treating imported target headers as system includes can change the header
+# search order if some targets have headers in the standard system include
+# directories, which can cause some headers not to be found due to the way
+# #include_next is used in the standard library.
+if(MINGW)
+	set(CMAKE_NO_SYSTEM_FROM_IMPORTED TRUE)
```

Comment:
Does this mean the include directories from `IMPORTED` targets will use `-I` instead of `-isystem`, like when we add directories with `include_directories` or  `target_include_directories` without `SYSTEM` option? Then why do we need to turn on this only in MinGW builds?

Reply:
That's exactly what it does. I've no idea why it's only needed for MinGW, as I would expect Linux builds to have the same problem. I guess the package headers and standard library headers are arranged differently. I'll investigate, but I don't think this is a problem for us.